packages/server/src/handlers/BaseHandler.ts

```diff
@@ -158,7 +162,81 @@ export class BaseHandler extends EventEmitter {
         }
       })
 
-      this.store.write(req.pipe(proxy), id, offset).then(resolve).catch(reject)
+      stream
+        .pipeline(req.pipe(proxy), new StreamLimiter(maxFileSize), async (stream) => {
```

Comment:
Can we add a comment why we use `req.pipe(proxy)` here instead of `pipeline(req, proxy, new SteamLimiter, ...)`? I know you answered this question in another PR, but it's good to document this in the code as well for future encounters.

Reply:
Done!